solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -525,8 +606,14 @@ private void constructState(Set<String> changedCollections) {
     Map<String, ClusterState.CollectionRef> result = new LinkedHashMap<>();
 
     // Add collections
-    for (Map.Entry<String, DocCollection> entry : watchedCollectionStates.entrySet()) {
-      result.put(entry.getKey(), new ClusterState.CollectionRef(entry.getValue()));
+    for (Map.Entry<String, DocCollectionWatch<DocCollectionWatcher>> entry :
+        collectionWatches.entrySet()) {
+      if (entry.getValue().currentDoc
+          != null) { // if the doc is null for the collection watch, then it should not be inserted
```

Comment:
Can you put the comment on the next line? That way the formatting of the if statement is preserved.